### ISSUE TITLE
First line is blank

### ISSUE THREAD

**Issue by thasmo** [jadejs/jade#1913]
Converting the following jade template with `gulp-jade` results in the HTML file having a blank, first line:
```js
link(rel='import', href='../../polymer/polymer.html')

polymer-element(name='my-element', attributes='attribute')
	template
	script(src='my-element.js')
```
Not sure if this depends on the source file or if this  is a general minor jade glitch, but it would be nice to get rid of the blank line.

Any ideas? Thanks!

**Comment by TimothyGu** [comment]
Unfortunately I do not think this is a bug in Jade. At least with Jade 1.9.2 it is not reproducible:

```sh
$ cat <<EOF | bin/jade.js 
link(rel='import', href='../../polymer/polymer.html')

polymer-element(name='my-element', attributes='attribute')
    template
    script(src='my-element.js')
EOF

<link rel="import" href="../../polymer/polymer.html"/><polymer-element name="my-element" attributes="attribute"><template></template><script src="my-element.js"></script></polymer-element>
```

**Comment by TimothyGu** [comment]
Scratch that.

**Comment by TimothyGu** [comment]
Hm WTF I can't reproduce it now.

**Comment by thasmo** [comment]
Hehe, I don't understand. ;) So, this may be more likely a gulp or gulp-jade bug?

**Comment by ForbesLindesay** [comment]
I'd be extremely surprised if this is a current jade bug.  This is almost certainly a bug in gulp-jade.  The other possibility is that you're using a very old version of jade, it used to be extremely buggy.  If you can reproduce it by calling jade directly (or via the CLI), please comment and we will re-open.

**Comment by finico** [comment]
@ForbesLindesay @thasmo I can reproduce it
```
> jade --version
1.11.0
> echo ".some" > ./template.jade
> jade ./template.jade --pretty

  rendered template.html

> cat template.html

<div class="some"></div>~/Desktop
> sed -n '$=' ./template.html
2
>
```
the same result on mac os and ubuntu server

**Comment by TimothyGu** [comment]
Seems like this issue is related to pretty mode.

**Comment by TimothyGu** [comment]
Yes I can reproduce it now. The problem is pretty difficult to fix, since we always insert a new line before a block-level tag (if you try `doctype html` or `span a` they don't have an extra new line). We need to add a flag in the Compiler object in jade-code-gen for determining whether or not we should add the new line, but this won't be accurate as we have mixins etc.
